gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -1298,31 +1328,34 @@ private Credential getCredential(
       throws IOException, GeneralSecurityException {
     Credential credential = null;
 
-    // Check if delegation token support is configured
-    if (delegationTokens != null) {
-      // If so, use the delegation token to acquire the Google credentials
-      AccessTokenProvider atp = delegationTokens.getAccessTokenProvider();
-      if (atp != null) {
-        atp.setConf(config);
+    // If the AuthorizationMode is set to `Request_Context_Relate`, Credential will be generated
+    // When GCS requests are created.
+    if (gcsFsOptions.getCloudStorageOptions().getAuthorizationMode()
```

Comment:
May we flip this condition and return `null` if it's default.

Also, we need to validate that all enums values are handled

Reply:
Sure, I will do both.

Reply:
FWIW, I didn't add a condition to throw for `else`. Let me know that should be there too.